src/pydrex/minerals.py

```diff
@@ -712,3 +633,120 @@ def from_file(cls, filename, postfix=None):
         mineral.fractions = fractions
         mineral.orientations = orientations
         return mineral
+
+
+def update_all(
+    minerals: list[Mineral],
+    params: dict,
+    deformation_gradient: np.ndarray,
+    get_velocity_gradient,
+    pathline: tuple,
+    get_regime=None,
+    **kwargs,
+) -> np.ndarray:
+    """Update orientations and volume distributions for all mineral phases.
+
+    Returns the updated deformation gradient tensor which measures the accumulated
+    macroscopic strain.
+
+    """
+    for i, mineral in enumerate(minerals):
+        if i == len(minerals) - 1:  # Collect new deformation gradient at last.
+            deformation_gradient = mineral.update_orientations(
+                params=params,
+                deformation_gradient=deformation_gradient,
+                get_velocity_gradient=get_velocity_gradient,
+                pathline=pathline,
+                get_regime=get_regime,
+                **kwargs,
+            )
+        else:
+            mineral.update_orientations(
+                params=params,
+                deformation_gradient=deformation_gradient,
+                get_velocity_gradient=get_velocity_gradient,
+                pathline=pathline,
+                get_regime=get_regime,
+                **kwargs,
+            )
```

Comment:
Not even sure you need the `if` here. Could just keep the upper block and update the comment to explain that the deformation gradient is independent of the mineral assemblage. And then rename the returned value to `updated_deformation_gradient`. Up to you.